File: gtmworks_ui/src/services/OpportunityService.js
import http from "../http-common"; 

class OpportunityService {
  getAllOpportunitys(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/opportunity/opportunitys`, searchDTO);
  }

  get(opportunityId) {
    return this.getRequest(`/opportunity/${opportunityId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/opportunity?field=${matchData}`, null);
  }

  addOpportunity(data) {
    return http.post("/opportunity/addOpportunity", data);
  }

  update(data) {
  	return http.post("/opportunity/updateOpportunity", data);
  }
  
  uploadImage(data,opportunityId) {
  	return http.postForm("/opportunity/uploadImage/"+opportunityId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new OpportunityService();

File: src/main/java/com/gtmworks/service/OpportunityService.java
package com.gtmworks.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.gtmworks.domain.Opportunity;
import com.gtmworks.dto.OpportunityDTO;
import com.gtmworks.dto.OpportunitySearchDTO;
import com.gtmworks.dto.OpportunityPageDTO;
import com.gtmworks.dto.OpportunityConvertCriteriaDTO;
import com.gtmworks.service.GenericService;
import com.gtmworks.dto.common.RequestDTO;
import com.gtmworks.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface OpportunityService extends GenericService<Opportunity, Integer> {

	List<Opportunity> findAll();

	ResultDTO addOpportunity(OpportunityDTO opportunityDTO, RequestDTO requestDTO);

	ResultDTO updateOpportunity(OpportunityDTO opportunityDTO, RequestDTO requestDTO);

    Page<Opportunity> getAllOpportunitys(Pageable pageable);

    Page<Opportunity> getAllOpportunitys(Specification<Opportunity> spec, Pageable pageable);

	ResponseEntity<OpportunityPageDTO> getOpportunitys(OpportunitySearchDTO opportunitySearchDTO);
	
	List<OpportunityDTO> convertOpportunitysToOpportunityDTOs(List<Opportunity> opportunitys, OpportunityConvertCriteriaDTO convertCriteria);

	OpportunityDTO getOpportunityDTOById(Integer opportunityId);







}





